Source/Laser/LaserEnvelope.cpp

```diff
@@ -0,0 +1,10 @@
+#include "LaserEnvelope.H"
```

Comment:
You need to add:
```
foreach(D IN LISTS WarpX_DIMS)
    warpx_set_suffix_dims(SD ${D})
    target_sources(WarpX_${SD}
      PRIVATE
        LaserEnvelope.cpp
    )
endforeach()
```
in this file: https://github.com/ECP-WarpX/WarpX/blob/development/Source/Laser/CMakeLists.txt

You will also need to add:
```
CEXE_sources += LaserEnvelope.cpp
```
in this file: https://github.com/ECP-WarpX/WarpX/blob/development/Source/Laser/Make.package